client/src/components/user/dashboard.js

```diff
@@ -83,20 +82,22 @@ class Dashboard extends React.Component {
 
   handleDashboardChange(newView) {
     const { mainComponentView } = this.state;
-
+    
     if (mainComponentView === 'dashboardCalendar' || mainComponentView === 'subscriptionDetail') {
-      this.setState({ mainComponentView: newView });
+      const newState = (newView === 'createSubscription') ? { mainComponentView: newView, activeSubscription: false } : { mainComponentView: newView };
```

Comment:
don't need parens here

```suggestion
      const newState = newView === 'createSubscription' ? { mainComponentView: newView, activeSubscription: false } : { mainComponentView: newView };
```